meta/webpmeta/webpmeta_test.go

```diff
@@ -0,0 +1,131 @@
+package webpmeta
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/mandykoh/prism/meta/binary"
+)
+
+func TestExtractMetadata(t *testing.T) {
+	t.Run("returns error with missing RIFF signature", func(t *testing.T) {
+		data := &bytes.Buffer{}
+		data.Write([]byte("NOT A RIFF SIGNATURE"))
+
+		_, err := extractMetadata(data)
+
+		if err == nil {
+			t.Errorf("Expected error but succeeded")
+		} else if expected, actual := "missing RIFF header", err.Error(); expected != actual {
+			t.Errorf("Expected error '%s' but got '%s'", expected, actual)
+		}
+	})
+
+	t.Run("returns error with missing WEBP signature", func(t *testing.T) {
+		data := &bytes.Buffer{}
+		data.Write([]byte("RIFF....NOTP"))
+
+		_, err := extractMetadata(data)
+
+		if err == nil {
+			t.Errorf("Expected error but succeeded")
+		} else if expected, actual := "not a WEBP file", err.Error(); expected != actual {
+			t.Errorf("Expected error '%s' but got '%s'", expected, actual)
+		}
+	})
+
+	t.Run("returns error if basic metadata is not found", func(t *testing.T) {
+		data := &bytes.Buffer{}
+		data.Write([]byte("RIFF\x28\x51\x04\x00WEBPVP8"))
+
+		_, err := extractMetadata(data)
+
+		if err == nil {
+			t.Errorf("Expected error but succeeded")
+		} else if expected, actual := "unexpected EOF reading chunk type", err.Error(); expected != actual {
```

Comment:
@mandykoh does it matter that the error is not "no metadata found"?

Reply:
No, I don’t think so. If we really need to differentiate this, I think it’d be better to introduce something that works with `errors.Is`.